### PR TITLE
Add serde prometheus

### PR DESCRIPTION
**Pull request by outscale-vbr**
This PR adds prometheus format output using serde format.

Closes https://github.com/outscale/osc-cost/issues/6

### CODE REVIEW

**Review by jerome-jutteau — APPROVED** [review]

**Comment by outscale-vbr** [comment]
Still in draft, it will be changed.

**Comment by outscale-vbr** [comment]
Remove from WIP.

File: src/ods/ser.rs [hunk 1]
```diff
@@ -311,7 +311,7 @@ impl<'a> ser::Serializer for &'a mut Serializer {
             sheet = self
                 .sheets
                 .get_mut(_name)
-                .ok_or(Error::Message("Internal error".to_string()))?
+                .ok_or_else(|| Error::Message("Internal error".to_string()))?
```
Comment (outscale-mdr) [inline src/ods/ser.rs:314]: Another commit :)

File: src/prometheus/ser.rs [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+            &_ => (),
+        }
+        match label_key {
+            label_key if *label_key == primary_key => gauge_state.primary_value = v.to_string(),
+            label_key if *label_key == secondary_key => gauge_state.secondary_value = v.to_string(),
+            label_key if *label_key == label_type => {
+                gauge_state.name = gauge_state.name.clone() + &v.to_string()
+            }
+            &_ => (),
+        }
```
Comment (outscale-mdr) [inline src/prometheus/ser.rs:118]: Is there a specific reason not to merge theses matches ?
```rust
match label_key {
            label_key if *label_key == primary_key => gauge_state.primary_value = v.to_string(),
            label_key if *label_key == secondary_key => gauge_state.secondary_value = v.to_string(),
            _ => {
                if label_key == label_type {
                   gauge_state.name = gauge_state.name.clone() + &v.to_string()
                }
                ...
}
```

File: src/prometheus/ser.rs [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+    gauge_primary_opt: Opts,
+    gauge_secondary_opt: Opts,
+    name: String,
+}
+pub fn to_prom<T>(
+    value: &T,
+    include: Vec<String>,
+    primary: CustomLabelKey,
+    secondary: CustomLabelKey,
+    label_type: String,
```
Comment (outscale-mdr) [inline src/prometheus/ser.rs:49]: label_name instead ?

File: src/prometheus/ser.rs [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+        let Ok(gauge_month) = Gauge::with_opts(gauge_state.gauge_secondary_opt.clone()) else {
+            return Err(Error::Message("Can not create gauge_secondary".to_string()))
+        };
+        gauge_hour.add(gauge_state.primary_value.parse::<f64>().unwrap_or_default());
+        gauge_month.add(
+            gauge_state
+                .secondary_value
+                .parse::<f64>()
+                .unwrap_or_default(),
+        );
```
Comment (outscale-mdr) [inline src/prometheus/ser.rs:382]: Do we need to add all const_labels before setting the value of the gauge ? 
If so, skip my comment.
If not, could we add the value of the gaguge during the serialization of the value ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]
Good job !

File: src/ods/ser.rs (hunk 1 above)
Comment (outscale-vbr) [reply src/ods/ser.rs:314]: Done

File: src/prometheus/ser.rs (hunk 2 above)
Comment (outscale-vbr) [reply src/prometheus/ser.rs]: Done

File: src/prometheus/ser.rs (hunk 3 above)
Comment (outscale-vbr) [reply src/prometheus/ser.rs]: Ok done

File: src/prometheus/ser.rs (hunk 4 above)
Comment (outscale-mdr) [reply src/prometheus/ser.rs]: Ok, nothing to do then

**Review by outscale-mdr — APPROVED** [review]
Good job !

**Pull request merged**